### DISCUSSION TITLE
getUserAppDataDir

### DISCUSSION THREAD

**Discussion by csb2024** [FreeCAD/FreeCAD#13025] [Code contribution discussions]
Can the function FreeCAD:: getUserAppDataDir() be defined as FreeCAD:: getHomePath()+"Data"? If defined this way, what problems will arise? When testing some third-party modules now, it is common to copy them directly to the FreeCAD:: getHomePath()+"Mod" location, but errors often occur. These modules often use os. path. abspath (__ file__) to obtain the current module path, which often leads to errors during runtime

**Comment by maxwxyz** [comment]
@chennes is this related to addon manager changes or general?

**Comment by chennes** [comment]
It's not really related to Addon Manager, the AM just puts things where FreeCAD tells it to.

**Comment by csb2024** [comment]
It's depends on the application,I change the code in my application
